static/images/diagrams/pxe_workflow.txt

```diff
@@ -0,0 +1,50 @@
+# Use http://sequencediagram.org/ to generate image
+
+title PXE-based provisioning with Puppet
+
+participant User
+participant "Provisioned Host" as Host
+participant "Foreman" as Foreman
+participant "Foreman Proxy" as Proxy
+participant TFTP
+participant DHCP
+participant DNS
+participant "Puppet master" as Puppet
+
+linear
+autonumber
+
+User->Foreman:clicks on New Host
+Foreman->Proxy:requests free IP address
+User->Foreman:clicks Submit button in the New host form
+Proxy->DHCP:forwards free IP request
```

Comment:
swap submit button and free IP request, the free IP is known before submit as it's shown in the form